src/misc.h

```diff
@@ -63,11 +63,20 @@ bool fileCopy(std::string source,std::string dest);
 std::string fileToBase64(std::string filepath);
 std::string fileGetMD5(std::string filepath);
 
-static inline bool strFind(std::string str, std::string target)
+static inline bool strFind(const std::string &str, const std::string &target)
 {
     return str.find(target) != str.npos;
 }
 
+static inline bool startsWith(const std::string &hay, const std::string &needle) {
+    return hay.substr(0, needle.length()) == needle;
+}
+
+static inline bool endsWith(const std::string &hay, const std::string &needle) {
+    size_t hl = hay.length(), nl = needle.length();
```

Comment:
Using `std::string::size_type` here might be better

Reply:
```suggestion
    std::string::size_type hl = hay.length(), nl = needle.length();
```